### PR TITLE
add glusterfs templates

### PR DESCRIPTION
**Pull request by LLParse**

### CODE REVIEW

File: templates/glusterfs/0/rancher-compose.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      label: "Gluster Networking Mode"
+      required: true
+      type: "enum"
+      default: "container:glusterfs-server"
+      options:
+        - "container:glusterfs-server"
+        - "host"
+glusterfs-server:
+  scale: 3
+  metadata:
```
Comment (cloudnautique) [inline templates/glusterfs/0/rancher-compose.yml]: Needs retain_ip: true

File: templates/glusterfs/0/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  cap_add:
+    - SYS_ADMIN
+  volumes_from:
+    - glusterfs-data
+  labels:
+    io.rancher.container.dns: 'true'
+    io.rancher.container.network: 'true'
+    io.rancher.container.hostname_override: container_name
+    io.rancher.scheduler.affinity:container_label_soft_ne: io.rancher.stack_service.name=$${stack_name}/glusterfs-server
+  command: glusterd -p /var/run/gluster.pid --no-daemon --log-file=-
```
Comment (cloudnautique) [inline templates/glusterfs/0/docker-compose.yml]: is this truncated?

Comment (LLParse) [inline templates/glusterfs/0/docker-compose.yml]: Yes, click 'View full changes' at the top right of the file. It's odd that github chose to truncate this file

File: templates/glusterfs/0/rancher-compose.yml (hunk 1 above)
Comment (LLParse) [inline templates/glusterfs/0/rancher-compose.yml]: fixed

File: templates/glusterfs/0/docker-compose.yml (hunk 2 above)
Comment (cloudnautique) [inline templates/glusterfs/0/docker-compose.yml]: Sorry, I though the `-` for a log file was an error. Good on you for figuring out thats how to get logs to stdout!

**Pull request merged**